spec/factory_bot/declaration_list_spec.rb

```diff
@@ -1,74 +1,72 @@
+    declaration_list = FactoryBot::DeclarationList.new
+
     allow(FactoryBot::AttributeList).to receive(:new).and_return attribute_list
 
-    subject.declare_attribute(declaration_1)
-    subject.declare_attribute(declaration_2)
+    declaration_list.declare_attribute(declaration1)
+    declaration_list.declare_attribute(declaration2)
 
-    subject.attributes
+    declaration_list.attributes
 
-    expect(attribute_list).to have_received(:define_attribute).with(attribute_1)
-    expect(attribute_list).to have_received(:define_attribute).with(attribute_2)
-    expect(attribute_list).to have_received(:define_attribute).with(attribute_3)
+    expect(attribute_list).to have_received(:define_attribute).with(attribute1)
+    expect(attribute_list).to have_received(:define_attribute).with(attribute2)
+    expect(attribute_list).to have_received(:define_attribute).with(attribute3)
   end
 end
 
 describe FactoryBot::DeclarationList, "#declare_attribute" do
-  let(:declaration_1) { double("declaration", name: "declaration 1") }
-  let(:declaration_2) { double("declaration", name: "declaration 2") }
-  let(:declaration_with_same_name) do
-    double("declaration", name: "declaration 1")
-  end
+  it "adds the declaration to the list when not overridable" do
+    declaration1 = double("declaration", name: "declaration 1")
+    declaration2 = double("declaration", name: "declaration 2")
+    declaration_list = FactoryBot::DeclarationList.new
 
-  context "when not overridable" do
-    it "adds the declaration to the list" do
-      subject.declare_attribute(declaration_1)
-      expect(subject.to_a).to eq [declaration_1]
+    declaration_list.declare_attribute(declaration1)
+    expect(declaration_list.to_a).to eq [declaration1]
 
-      subject.declare_attribute(declaration_2)
-      expect(subject.to_a).to eq [declaration_1, declaration_2]
-    end
+    declaration_list.declare_attribute(declaration2)
+    expect(declaration_list.to_a).to eq [declaration1, declaration2]
   end
 
-  context "when overridable" do
-    before { subject.overridable }
+  it "adds the declaration to the list when overridable" do
+    declaration1 = double("declaration", name: "declaration 1")
+    declaration2 = double("declaration", name: "declaration 2")
+    declaration_list = FactoryBot::DeclarationList.new
+    declaration_list.overridable
 
-    it "adds the declaration to the list" do
-      subject.declare_attribute(declaration_1)
-      expect(subject.to_a).to eq [declaration_1]
+    declaration_list.declare_attribute(declaration1)
+    expect(declaration_list.to_a).to eq [declaration1]
+
+    declaration_list.declare_attribute(declaration2)
+    expect(declaration_list.to_a).to eq [declaration1, declaration2]
+  end
 
-      subject.declare_attribute(declaration_2)
-      expect(subject.to_a).to eq [declaration_1, declaration_2]
-    end
+  it "deletes declarations with the same name when overridable" do
```

Comment:
It seems to me there is a test case missing here: `it "adds declarations with the same name to the list when NOT overridable"`. Do you want to add it?

Reply:
Done.  I *kinda* wish the `.overridable?` method was public instead of private, so I could assert against it in this new test and make the intention a bit more clear.  Unfortunately that's not really advisable if that public method only gets called in the test suite.